PortalForecasts.R

```diff
@@ -180,35 +188,42 @@ forecastall <- function(abundances,level,weather_data,weatherforecast) {
 
     if(sum(species_abundance) == 0){
       pred = zero_abund_forecast
+      model_aic = 1e6
     } else {
       best_model_aic = Inf
       best_model = NA
       for(proposed_model_covariates in model_covariates){
         proposed_model = tsglm(species_abundance,model=list(past_obs=1,past_mean=12),distr="poisson",xreg=weather_data[,unlist(proposed_model_covariates)],link = "log")
         #tsglm sometimes outputs an error when the time series have many 0's, in that case set the AIC
-        #to Inf to this proposed model covariate set get skipped
+        #to Inf so this proposed model covariate set get skipped
         proposed_model_aic = ifelse(has_error(summary(proposed_model))==T,Inf,summary(proposed_model)$AIC)
         if(proposed_model_aic < best_model_aic){
           best_model = proposed_model
           best_model_aic = proposed_model_aic
         }
       }
-    }
-    #If no best model was chosen, ie. they all had infinit AIC's due to errors in model building
-    #then forecast 0's
-    if(is.na(best_model)){
-      pred = zero_abund_forecast
-    } else {
-      pred = predict(best_model,12,level=0.9,newdata=weathermeans) 
-    }
     
+      #If no best model was chosen, ie. they all had infinit AIC's due to errors in model building
+      #then forecast 0's. Also make an extremely high AIC so this isn't weighted heavily in the ensemble. 
+      if(is.na(best_model)){
+        pred = zero_abund_forecast
+        model_aic = 1e6
+      } else {
+        pred = predict(best_model,12,level=0.9,newdata=weathermeans)
```

Comment:
There are a lot of 12's and 0.9's in the code (or occasionally 90 instead of 0.9). It would be useful to give these values a name and control them from a single place (e.g. as arguments to the function). That would make it much easier if we decide some day that we want to predict 24 months into the future or want 95% CIs.